lib/vagrant-openshift/action/install_origin_base_dependencies.rb

```diff
@@ -131,19 +110,73 @@ def call(env)
 chmod 755 /usr/bin/chromedriver
           }, :timeout=>60*60, :verbose => false)
 
-          sudo(env[:machine], "wget -O /etc/yum.repos.d/openshift-rhel7-dependencies.repo https://mirror.openshift.com/pub/openshift-origin/nightly/rhel-7/dependencies/openshift-rhel7-dependencies.repo", fail_on_error: true, :timeout=>60*20, :verbose => true)
 
-          #
-          # FIXME: Need to install golang packages 'after' the 'gcc' is
-          #        installed. See BZ#1101508
-          #
-          sudo(env[:machine], "yum install -y golang golang-src", :timeout=>60*20, :verbose => false)
-          #
+          # We want to allow users to provide RPM repositories as well as names of RPM repositories
+          # from which Docker and Golang dependencies are installed. In order to sandbox the process
+          # to make it difficult to propagate unwanted consequences, we will approach the install by:
+          #   - installing all of the repositories that the user provided
+          #   - installing the default (non-standard RHEL) repositories that provide Docker and Golang
+          #   - installing Golang and Docker with the versions provided by only looking at the explicitly
+          #     enabled repositories
+          #   - disabling all of the repositories that we installed previously
+
+          # install user-provided repositories
+          @options[:repofiles].each do |path|
+            repospec = File.open(path, "r").read
 
-          unless is_fedora
-            sudo(env[:machine], "yum install -y golang-pkg-linux-amd64", :timeout=>60*20, :verbose => false)
+            reponame = repospec.lines.first.tr("[]", "")
+            @options[:reponames] << reponame
+
+            repofile = "/etc/yum.repos.d/#{reponame}.repo"
+            File.open(repofile, "w").write(repospec)
           end
 
+          # if we're on RHEL, install `dockerextra`, `origin-deps-rhel7` and `origin-deps-rhel7-source`
+          # default repositories for providing newer versions of Docker and Golang than the normal
+          # RHEL RPM streams
+          if env[:machine].communicate.test("test -e /etc/redhat-release && ! test -e /etc/fedora-release && ! test -e /etc/centos-release")
+            File.open("/etc/yum.repos.d/dockerextra.repo", "w").write(%{
+[dockerextra]
+name=RHEL Docker Extra
+baseurl=https://mirror.openshift.com/enterprise/rhel/dockerextra/x86_64/os/
+enabled=1
+gpgcheck=0
+failovermethod=priority
+sslverify=False
+sslclientcert=/var/lib/yum/client-cert.pem
+sslclientkey=/var/lib/yum/client-key.pem
+              })
+            @options[:reponames] << "dockerextra"
+
+            File.open("/etc/yum.repos.d/origin-deps-rhel7.repo", "w").write(%{
```

Comment:
I don't see a good reason to be fetching this file from the web each time, but we can keep that. I like this approach as it's closer to the other hard-coded repo and because converting the `wget` to Ruby would mean we `require 'open-uri'` above and I'm not sure if that is totally ok (will that dep always be present?).